lib/matplotlib/cm.py

```diff
@@ -231,50 +231,6 @@ def get_cmap(self, cmap):
 globals().update(_colormaps)
 
 
-@_api.deprecated("3.7", alternative="``matplotlib.colormaps.register(name)``")
-def register_cmap(name=None, cmap=None, *, override_builtin=False):
-    """
-    Add a colormap to the set recognized by :func:`get_cmap`.
-
-    Register a new colormap to be accessed by name ::
-
-        LinearSegmentedColormap('swirly', data, lut)
-        register_cmap(cmap=swirly_cmap)
-
-    Parameters
-    ----------
-    name : str, optional
-       The name that can be used in :func:`get_cmap` or :rc:`image.cmap`
-
-       If absent, the name will be the :attr:`~matplotlib.colors.Colormap.name`
-       attribute of the *cmap*.
-
-    cmap : matplotlib.colors.Colormap
-       Despite being the second argument and having a default value, this
-       is a required argument.
-
-    override_builtin : bool
-
-        Allow built-in colormaps to be overridden by a user-supplied
-        colormap.
-
-        Please do not use this unless you are sure you need it.
-    """
-    _api.check_isinstance((str, None), name=name)
-    if name is None:
-        try:
-            name = cmap.name
-        except AttributeError as err:
-            raise ValueError("Arguments must include a name or a "
-                             "Colormap") from err
-    # override_builtin is allowed here for backward compatibility
-    # this is just a shim to enable that to work privately in
-    # the global ColormapRegistry
-    _colormaps._allow_override_builtin = override_builtin
```

Comment:
The handling of `_allow_override_builtin` should be removed from `ColormapRegistry` as this is the only place that set it to `True`.

Reply:
I've removed the shim and updated the logic for `def register` for `ColormapRegistry` 
```python
            elif name in self._builtin_cmap:
                # We don't allow overriding a builtin
                raise ValueError("Re-registering the builtin cmap "
                                 f"{name!r} is not allowed.")
```
I updated the conditional to remove the override_builtin condition. Does this look ok or should the entire conditional be removed?

Reply:
I'm not seeing that change here?

Reply:
I see it now; that's the right change here.